tools/pipelines/test-service-clients.yml

```diff
@@ -88,3 +88,45 @@ stages:
           odsp__client__login__password: $(odsp-client-login-password)
           odsp__client2__login__username: $(odsp-client2-login-username)
           odsp__client2__login__password: $(odsp-client2-login-password)
+
+  # Capture pipeline stage results
+  - ${{ if eq(parameters.telemetry, true) }}:
+    - stage: upload_run_telemetry
+      displayName: Upload pipeline run telemetry to Kusto
+      condition: succeededOrFailed()
+      dependsOn:
+        - e2e - azure client with frs
+        - e2e - azure client with azure local service
+        - e2e - odsp client with odsp service
+      jobs:
+        - job: upload_run_telemetry
+          displayName: Upload pipeline run telemetry to Kusto
+          pool: Small
+          variables:
+          - group: ado-feeds
+    
+          steps:
+          - template: templates/include-telemetry-setup.yml
+            parameters:
+              devFeedUrl: $(ado-feeds-dev)
+              officeFeedUrl: $(ado-feeds-office)
+              isCheckoutNeeded: true
+          - task: Bash@3
+            displayName: Retrieve buildId results
+            inputs:
+              targetType: 'inline'
+              workingDirectory: $(absolutePathToTelemetryGenerator)
```

Comment:
We need to add [the variable definition](https://github.com/microsoft/FluidFramework/blob/218b985f5708c781ee31faaf639209c2a0de0fc6/tools/pipelines/templates/1ES/build-npm-package.yml#L646-L648) to the job's `variables` section.